### ISSUE TITLE
SequentialLR can raise bug when using pytorch_lightning

### ISSUE THREAD

**Issue by hehyuan** [pytorch/pytorch#80291]
### 🐛 Describe the bug

```python
# when scheduler choose SequentialLR
lrs.SequentialLR(optimizer, schedulers=[warmup_scheduler, step_scheduler], milestones=[self.hparams.warm_up_iter])
...

# using pytorch_ligntning to train model, then return error because SequentialLR has no attribute 'optimizer'

model.fit()
```

AttributeError: 'SequentialLR' object has no attribute 'optimizer'

### Versions

my pytorch version is '1.10.0' and python version is 3.8.0, pytorch_lightning version is '1.5.0'

cc @vincentqb @jbschlosser @albanD

**Comment by hehyuan** [comment]
I think add 
```python
self.optimizer = optimizer
``` 
in __init__() method of SequentialLR can be help, but there might cause other problems.

**Comment by jbschlosser** [comment]
I believe this was addressed in #67406. @hehyuan please feel free to reopen if you find ongoing issues in a nightly build of PyTorch.
